apps/cnquery/cmd/login.go

```diff
@@ -63,10 +62,12 @@ func register(token string) {
 	apiEndpoint := viper.GetString("api_endpoint")
 	token = strings.TrimSpace(token)
 
-	httpClient, err := httpclient.NewClient()
+	// NOTE: login is special because we do not have a config yet
+	proxy, err := cnquery_config.GetAPIProxy()
```

Comment:
Avoiding reading the config file means the ansible and chef stuff will need to work a different way. I suppose we can just do something like:

`if API_PROXY != nil {
  cnspec login --api-proxy=API_PROXY ...
}

if API_PROXY == nil {
  cnspec login ...
}`

in the ansible/chef logic paths

Reply:
Because it was reading the config file, and it non-destructively adds the extra fields, it was/is an easy way to set the api_proxy and have it stick around after `login`

Reply:
Although, thinking about it, it is a bit surprising that `login` reads the config file. Perhaps this behavior is less surprising, and we can just add the `--api-proxy` param as part of `login`, and add the yaml `api_proxy` field right after successful login.